src/main/kotlin/threed/example/DrawTwoRectangles.kt

```diff
@@ -0,0 +1,155 @@
+    val vertices = arrayOf(
+            -1.0f, 1.0f, 0.0f,
+            -1.0f, -1.0f, 0.0f,
+            1.0f, -1.0f, 0.0f,
+            1.0f, 1.0f, 0.0f
+    )
+
+    //
+    val indices = arrayOf<Short>(
+            0, 1, 2,
+            2, 3, 0
+    )
+
+    val colors = arrayOf(
+            color.red, color.green, color.blue, color.alpha,
+            color.red, color.green, color.blue, color.alpha,
+            color.red, color.green, color.blue, color.alpha,
+            color.red, color.green, color.blue, color.alpha,
+            color.red, color.green, color.blue, color.alpha,
+            color.red, color.green, color.blue, color.alpha
+    )
+
+    val indexBuffer = gl.createBuffer()
+    gl.bindBuffer(ELEMENT_ARRAY_BUFFER, indexBuffer)
+    gl.bufferData(ELEMENT_ARRAY_BUFFER, Uint16Array(indices), STATIC_DRAW)
+
+    val colorBuffer = gl.createBuffer()
+    gl.bindBuffer(ARRAY_BUFFER, colorBuffer)
+    gl.bufferData(ARRAY_BUFFER, Float32Array(colors), STATIC_DRAW)
+
+    // Create and store data into vertex buffer
+    val vertexBufferRectangle = gl.createBuffer()
+    gl.bindBuffer(ARRAY_BUFFER, vertexBufferRectangle)
+    gl.bufferData(ARRAY_BUFFER, Float32Array(vertices), STATIC_DRAW) // put the rectangle coordinates into the current array buffer
+
+
+    gl.bindBuffer(ARRAY_BUFFER, vertexBufferRectangle)
+    val verticesAttribute = gl.getAttribLocation(shaderProgram, "vertices") // position is the name of the attribute in the vertex shader code
+    gl.vertexAttribPointer(verticesAttribute, 3, FLOAT, false, 0, 0)
+    gl.enableVertexAttribArray(verticesAttribute)
+
+    gl.bindBuffer(ARRAY_BUFFER, colorBuffer)
+    val colorAttribute = gl.getAttribLocation(shaderProgram, "color")
+    gl.vertexAttribPointer(colorAttribute, 4, FLOAT, false, 0, 0)
+    gl.enableVertexAttribArray(colorAttribute)
+
+    gl.useProgram(shaderProgram)
+
+    val modeMatrix =
+            translateMatrix(transform.x, transform.y, transform.z) *
+                    rotateXMatrix(0.0.asRad.toFloat()) *
+                    rotateYMatrix(0.0.asRad.toFloat()) *
+                    rotateZMatrix(0.0.asRad.toFloat()) *
+                    scaleMatrix(scale.x, scale.y, scale.z)
+
+
+    val modelMatrixUniform = gl.getUniformLocation(shaderProgram, "modelMatrix")
+    gl.uniformMatrix4fv(modelMatrixUniform, false, modeMatrix)
+
+    gl.bindBuffer(ELEMENT_ARRAY_BUFFER, indexBuffer)
+    gl.drawElements(TRIANGLES, indices.size, UNSIGNED_SHORT, 0)
+
+
+}
+
+fun drawTwoRectangles(gl: WebGLRenderingContext) {
+
+    val rectangleShaderProgram = createRectangleShaderProgram(gl)
+
+    gl.clearColor(0.9f, 0.9f, 0.9f, 1.0f)
+    gl.clear(COLOR_BUFFER_BIT)
+
+    drawRectangle(gl, rectangleShaderProgram,
+            color = Rgba(1.0f, 0.0f, 0.0f),
+            transform = Coordinate(-0.5f, -0.5f, 0.0f),
+            scale = Coordinate(0.5f, 0.5f, 0.0f)
+    )
+
+    drawRectangle(gl, rectangleShaderProgram,
```

Comment:
Es tut grundsätzlich was es soll, ABER:
Das Rectangle wird zwei Mal in den Grafikspeicher geladen -> besser nur einmal Laden und dann mit der Model Matrix transformiert, mehrmals zeichnen. (so wie ich es mit den drawMultipleCubes gemacht habe)
Das gleiche sollte auch mit dem Fragment Shader funktionieren, aber wir haben uns ja vorerst einmal auf einfarbige Formen geeinigt.